custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -219,6 +227,117 @@ def handle_exception(self, exception, repeat_record):
                                 "hiv_test_nikshay_registered", "hiv_test_nikshay_error")
 
 
+@RegisterGenerator(NikshayFollowupRepeater, 'case_json', 'JSON', is_default=True)
+class NikshayFollowupPayloadGenerator(BaseNikshayPayloadGenerator):
+    def get_payload(self, repeat_record, test_case):
+        self.test_case = test_case
+        occurence_case = get_occurrence_case_from_test(test_case.domain, test_case.get_id)
+        episode_case = get_open_episode_case_from_occurrence(test_case.domain, occurence_case.get_id)
+        person_case = get_person_case_from_occurrence(test_case.domain, occurence_case.get_id)
+
+        test_case_properties = test_case.dynamic_case_properties()
+        episode_case_properties = episode_case.dynamic_case_properties()
+
+        interval_id, lab_serial_number, result_grade, dmc_code = self._get_mandatory_fields(test_case_properties)
+
+        test_conducted_on = _format_date(test_case_properties, 'date_tested')
+        properties_dict = self._base_properties(repeat_record)
+        properties_dict.update({
+            "PatientID": episode_case_properties.get('nikshay_id'),
+            "TestDate": test_conducted_on,
+            "LabNo": lab_serial_number,
+            "Local_ID": person_case.get_id,
+            "IntervalId": interval_id,
+            # since weight is not taken and is mandatory we send 1
```

Comment:
The reason we have this as "1" because weight is not collected on each follow up so it was decided to not mention it at all instead of having the same value from episode each time and since its mandatory we send 1.